velox/exec/tests/ContainerRowSerdeTest.cpp

```diff
@@ -62,6 +95,31 @@ class ContainerRowSerdeTest : public testing::Test,
     allocator_.clear();
   }
 
+  void testCompareWithNulls(
+      const DecodedVector& decodedVector,
+      const std::vector<HashStringAllocator::Position>& positions,
+      const std::vector<std::shared_ptr<ByteStream>>& byteStreams,
+      const std::vector<int32_t>& expected,
+      std::optional<int32_t> expectedOptionVal,
+      CompareFlags compareFlags) {
+    for (auto i = 0; i < expected.size(); ++i) {
+      ASSERT_EQ(
+          expected.at(i),
+          ContainerRowSerde::compareWithNulls(
+              *byteStreams.at(i), decodedVector, i, compareFlags)
+              .value());
+    }
+
+    auto res = ContainerRowSerde::compareWithNulls(
+        *byteStreams.at(3), decodedVector, 3, compareFlags);
```

Comment:
Hard-coding index '3' here makes this method hard to use. Let's change expected to std::vector<std::optional<int32_t>> and remove expectedOptionVal.

Reply:
Fixed